### DISCUSSION TITLE
OpenID/Oauth

### DISCUSSION THREAD

**Discussion by autarchprinceps** [authelia/authelia#7141] [Q&A] [answered]
Has anybody actually got any kind of sample of a working openid/oauth configuration for authelia?
I have a working basic Authelia, now I'm trying to switch to GitHub Oauth as identity provider instead, but while I have gotten it to stop erroring, it still asks for the normal user and just logs in after providing that, with virtually no changes due to the addition of oidc.
The documentation is sadly more than cryptic and I have no idea what else I could configure to make it use oidc.

**Comment by glime-ai** [comment, marked as answer]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Based on the context provided, it seems you are trying to configure Authelia to use GitHub OAuth as an identity provider. However, according to the information from the GitHub discussion (https://github.com/authelia/authelia/discussions/4471), Authelia's roadmap for OpenID Connect 1.0 is focused on it being a provider, not a client. This means that Authelia is designed to serve as the identity provider itself, rather than delegating authentication to an external provider like GitHub.

The discussion explicitly mentions that the current roadmap for OpenID Connect 1.0 within Authelia does not include functionality for acting as a client to external OAuth2 or similar providers. This would be necessary for implementing a "login with GitHub" type of functionality. The roadmap indicates that this might be considered in the future, but as of the last update provided in your context, it is not yet a feature.

Therefore, if you are trying to use GitHub OAuth as an identity provider with Authelia, it appears that this is not currently supported. Authelia's focus is on being the identity provider, where it manages the authentication process directly rather than integrating with external identity providers like GitHub for OAuth.

For your use case, you might need to look into other solutions or workarounds that could involve direct integration with GitHub OAuth for your application, bypassing Authelia for the GitHub authentication part, or wait until Authelia potentially adds support for acting as an OAuth client in the future.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
Not sure why the downvote, the answer is mostly correct. We have plans for it but it's not currently implemented.

**Reply by sanjudhyan** [reply]
Hi @james-d-elliott , I am trying to use google sso with authelia, i am uisng the below code to use google sso, but its erroring out saying that 
msg="Configuration: configuration key not expected: identity_providers.external.openid_connect.providers.google.id" and for most of the fields which is there inside identity_providers
identity_providers:
  oidc:
    hmac_secret: "super_secret_change_me"
  external:
    openid_connect:
      providers:
        google:
          id: google
          label: "Sign in with Google"
          client_id: "YOUR_GOOGLE_CLIENT_ID"
          client_secret: "YOUR_GOOGLE_CLIENT_SECRET"
          discovery_url: "https://accounts.google.com/.well-known/openid-configuration"
          scopes:
            - openid
            - email
            - profile
can you please confirm is this supprted in the latest version of authelia i.e. 4.39.14
